### PR TITLE
Added new Java Course

### PR DESCRIPTION
**Pull request by Nandan-18**
## What does this PR do?
Add resource(s) | Add info | Improve repo

### Description

Free Java course offered by Sololearn.

### Why is this valuable (or not)?

Platform has an interactive and clean interface.

### How do we know it's really free?

I've used it myself to learn concepts.

### For book lists, is it a book? For course lists, is it a course? etc.

It is a course.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
add an access note indicating registration required

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -393,6 +393,7 @@
 * [Java Programming: Solving Problems with Software](https://www.coursera.org/learn/java-programming) - (Coursera)
 * [Java tutorial for beginners](https://youtube.com/playlist?list=PLZPZq0r_RZOMhCAyywfnYLlrjiVOkdAI1) - Bro Code (YouTube)
 * [Learn how to program: Java](https://www.learnhowtoprogram.com/java-june-2017) - Epicodus Inc.
+* [Learn Java](https://www.sololearn.com/learning/1068) - (Sololearn) (Registration Required)
```
Comment (davorpa) [inline courses/free-courses-en.md:396]: Following [CONTRIBUTING format guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#formatting):
>- if an email address or account setup is requested before download is enabled, add language-appropriate notes in parentheses, e.g.: (email address *requested*, not required)

Maybe better

```markdown
* [Learn Java](https://www.sololearn.com/learning/1068) - Sololearn *(email address required)*
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 1 above)
Comment (eshellman) [reply courses/free-courses-en.md:396]: yes

**Review by davorpa — APPROVED** [review]
Sololearn, a good platform with courses for beginners, code playgrounds and interesting challenges. Maybe, Q&A and competitive community would be the best of it.

### Ready to merge :fireworks: :tada:

**Pull request merged**